wicket/src/cli/inventory.rs

```diff
@@ -0,0 +1,110 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Support for inventory checks via wicketd.
+
+use crate::cli::CommandOutput;
+use crate::wicketd::create_wicketd_client;
+use anyhow::Context;
+use anyhow::Result;
+use clap::Subcommand;
+use owo_colors::OwoColorize;
+use sled_hardware_types::Baseboard;
+use slog::Logger;
+use std::net::SocketAddrV6;
+use std::time::Duration;
+use wicket_common::rack_setup::BootstrapSledDescription;
+
+const WICKETD_TIMEOUT: Duration = Duration::from_secs(5);
+
+#[derive(Debug, Subcommand)]
+pub(crate) enum InventoryArgs {
+    /// List state of all bootstrap sleds, as configured with rack-setup
+    ConfiguredBootstrapSleds {
+        /// Print output as json
+        #[clap(long)]
+        json: bool,
+    },
+}
+
+impl InventoryArgs {
+    pub(crate) async fn exec(
+        self,
+        log: Logger,
+        wicketd_addr: SocketAddrV6,
+        mut output: CommandOutput<'_>,
+    ) -> Result<()> {
+        let client = create_wicketd_client(&log, wicketd_addr, WICKETD_TIMEOUT);
+
+        match self {
+            InventoryArgs::ConfiguredBootstrapSleds { json } => {
+                // We don't use the /bootstrap-sleds endpoint, because that
+                // gets all sleds visible on the bootstrap network. We want
+                // something subtly different here.
+                // - We want the status of only sleds we've configured wicket
+                //   to use for setup. /bootstrap-sleds will give us sleds
+                //   we don't want
+                // - We want the status even if they aren't visible on the
+                //   bootstrap network yet.
+                //
+                // In other words, we want the sled information displayed at the
+                // bottom of the rack setup screen in the TUI, and we get it the
+                // same way it does.
+                let conf = client
+                    .get_rss_config()
+                    .await
+                    .context("failed to get rss config")?;
+
+                let bootstrap_sleds = &conf.insensitive.bootstrap_sleds;
+                if json {
+                    let json_str = serde_json::to_string(bootstrap_sleds)
```

Comment:
nitpicky, take it or leave it: maybe nicer to use 
```suggestion
                    let json_str = serde_json::to_string_pretty(bootstrap_sleds)
```
here, since there isn't a ton of JSON and we're not trying to send it over the network? that way, a human user gets more readable output by default, which is maybe nice if the thing consuming the JSON chokes on it?

Reply:
Yeah, love the idea of using `to_string_pretty`.